### PR TITLE
Create a metric to track users with newlines in ingested label values

### PR DESCRIPTION
**Pull request by colega**
#### What this PR does

Creates an experimental, most likely temporary, metric `cortex_distributor_label_values_with_newlines_total` used to track tenants who are sending newlines in their label values.

#### Which issue(s) this PR fixes or relates to

Fixes an internal request

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: pkg/distributor/distributor.go [hunk 1]
```diff
@@ -1053,10 +1069,12 @@ func (d *Distributor) prePushValidationMiddleware(next PushFunc) PushFunc {
 
 			validatedSamples += len(ts.Samples) + len(ts.Histograms)
 			validatedExemplars += len(ts.Exemplars)
+			labelValuesWithNewlines += d.labelValuesWithNewlines(ts.Labels)
 		}
 
 		d.incomingSamplesPerRequest.WithLabelValues(userID).Observe(float64(totalSamples))
 		d.incomingExemplarsPerRequest.WithLabelValues(userID).Observe(float64(totalExemplars))
+		d.labelValuesWithNewlinesPerUser.WithLabelValues(userID).Add(float64(labelValuesWithNewlines))
```
Comment (pstibrany) [inline pkg/distributor/distributor.go:1077]: can we only add if its > 0? (No need to track users with 0 such labels, right?)

File: pkg/distributor/distributor.go [hunk 2]
```diff
@@ -412,6 +413,10 @@ func New(cfg Config, clientConfig ingester_client.Config, limits *validation.Ove
 			Name: "cortex_distributor_latest_seen_sample_timestamp_seconds",
 			Help: "Unix timestamp of latest received sample per user.",
 		}, []string{"user"}),
+		labelValuesWithNewlinesPerUser: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Name: "cortex_distributor_label_values_with_newlines_total",
+			Help: "Total amount of label values with newlines seen at ingestion time.",
```
Comment (pstibrany) [inline pkg/distributor/distributor.go:418]: nit: amount -> number. "amount" is used for uncountable items.

File: pkg/distributor/distributor_test.go [hunk 3]
```diff
@@ -374,6 +374,7 @@ func TestDistributor_MetricsCleanup(t *testing.T) {
 	d.nonHASamples.WithLabelValues("userA").Add(5)
 	d.dedupedSamples.WithLabelValues("userA", "cluster1").Inc() // We cannot clean this metric
 	d.latestSeenSampleTimestampPerUser.WithLabelValues("userA").Set(1111)
+	d.labelValuesWithNewlinesPerUser.WithLabelValues("userA").Inc()
```
Comment (pstibrany) [inline pkg/distributor/distributor_test.go:377]: this change does nothing, unless we ask for new metric too (add `cortex_distributor_label_values_with_newlines_total` to `metrics`)

**Review by pstibrany — APPROVED** [review]
lgtm, with minor comments. thanks.

File: pkg/distributor/distributor.go (hunk 2 above)
Comment (colega) [reply pkg/distributor/distributor.go]: Done in https://github.com/grafana/mimir/pull/9400/commits/faa9ea4b6bdb888d600fb80d195885fe27d332e3

File: pkg/distributor/distributor_test.go (hunk 3 above)
Comment (colega) [reply pkg/distributor/distributor_test.go:378]: Thanks, made the test meaningful in https://github.com/grafana/mimir/pull/9400/commits/0bed651811575e74e4d58d3c51c89b9720242ee6

**Review by pstibrany — APPROVED** [review]
Thank you.

**Pull request merged**